crates/pypi-types/src/simple_json.rs

```diff
@@ -37,14 +37,17 @@ fn sorted_simple_json_files<'de, D: Deserializer<'de>>(d: D) -> Result<Vec<File>
 #[derive(Debug, Clone, Deserialize)]
 #[serde(rename_all = "kebab-case")]
 pub struct File {
-    // Non-PEP 691-compliant alias used by PyPI.
-    #[serde(alias = "data-dist-info-metadata")]
+    // PEP 714-renamed field, followed by PEP 691-compliant field, followed by non-PEP 691-compliant
+    // alias used by PyPI.
+    pub core_metadata: Option<DistInfoMetadata>,
     pub dist_info_metadata: Option<DistInfoMetadata>,
+    pub data_dist_info_metadata: Option<DistInfoMetadata>,
```

Comment:
Repeating like this is the only way I could figure out to support multiple aliases under different names, where _multiple_ values can be present. (PyPI serves both `core-metadata` and `data-dist-info-metadata`; serde's ` alias` feature errors if multiple values are present.)

Reply:
Per PEP 714, we _could_ omit `data_dist_info_metadata`... But there may be registries that _only_ serve that in practice, since that's what pip did for a while.

Reply:
I would define a separate wire type with what you want to support, and then implement `From<FileWire> for File` (or `TryFrom` if needed). Then use [`#[serde(from = "FileWire")]`](https://serde.rs/container-attrs.html#from). That way, you can resolve the aliases down to one field at deserialization time, and avoid spreading that complexity outward to anyone consuming `File`.

Reply:
Hmm, but we already have a separate `File` type that we use in our own abstractions.

Reply:
So in a sense this _is_ `FileWire`. And we have a `From` elsewhere.

Reply:
Yeah this is meant to be roughly "the exact JSON we get from the API", so I suppose this is ok.